File: backend/ModuMessenger/src/main/java/com/example/modumessenger/common/exception/ErrorCode.java
package com.example.modumessenger.common.exception;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

import static org.springframework.http.HttpStatus.*;

@Getter
@RequiredArgsConstructor
public enum ErrorCode {

    // 200
    SUCCESS(OK, "OK"),

    // 400
    NOT_SUPPORTED_HTTP_METHOD(BAD_REQUEST,"지원하지 않는 Http Method 방식입니다."),
    NOT_VALID_METHOD_ARGUMENT(BAD_REQUEST,"유효하지 않은 Request Body 혹은 Argument 입니다."),
    EMAIL_NOT_FOUND(BAD_REQUEST, "email을 찾을 수 없습니다."),
    USERID_NOT_FOUND(BAD_REQUEST, "해당 사용자를 찾을 수 없습니다."),

    // 401
    UNAUTHORIZED_TOKEN_ERROR(UNAUTHORIZED, "인증되지 않은 토큰 입니다.");

    private final HttpStatus status;
    private final String message;
}

File: backend/point-service/src/main/kotlin/com/example/pointservice/common/exception/ErrorCode.kt
package com.example.pointservice.common.exception

import org.springframework.http.HttpStatus

enum class ErrorCode(val status: HttpStatus, val message: String) {
    INVALID_AMOUNT(HttpStatus.BAD_REQUEST, "포인트 금액이 올바르지 않습니다."),
    RULE_NOT_FOUND(HttpStatus.NOT_FOUND, "적립 규칙을 찾을 수 없습니다."),
    RULE_ALREADY_EXISTS(HttpStatus.CONFLICT, "같은 코드의 적립 규칙이 이미 있습니다."),
    RULE_IN_USE(HttpStatus.CONFLICT, "출석 체크가 쓰는 규칙이라 지울 수 없습니다. 대신 비활성으로 바꾸세요."),
    ACCOUNT_NOT_FOUND(HttpStatus.NOT_FOUND, "포인트 계정을 찾을 수 없습니다."),
    INSUFFICIENT_POINT(HttpStatus.CONFLICT, "포인트가 부족합니다."),
}
